### ISSUE TITLE
Request: Validate toolchain

### ISSUE THREAD

**Issue by vanyauhalin** [koalaman/shellcheck#3116]
Hello Vidar! I have a request. Will SpellCheck be able to check not only script syntax in the future but also the commands used in the script? My problem is that I like to use long flags in my scripts, but when I need to write something for BusyBox, I often forget that it does not support them in common. It would be great if this could be tracked in advance. Thanks!

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/busybox sh

echo "a b" | cut --delimiter " " --fields 2 # GNU cut
echo "a b" | cut -d" " -f2                  # BusyBox cut
```

#### Here's what shellcheck currently says:

```txt
nothing
```

#### Here's what I wanted or expected to see:

```txt
BusyBox cut command does not support --delimiter option, replace it with -d.
BusyBox cut command does not support --fields option, replace it with -f.
```

**Comment by Ordoviz** [comment]
Related: #2758
